packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -118,9 +118,24 @@ class CodeSizeController {
     changeDiffRoot(_activeDiffRoot);
   }
 
-  void loadTreeFromJsonFile(DevToolsJsonFile jsonFile) {
+  /// Notifies that the json files are currently being processed.
+  ValueListenable get processingNotifier => _processingNotifier;
+  final _processingNotifier = ValueNotifier<bool>(false);
+
+  // TODO(peterdjlee): Spawn an isolate to run parts of this function to
+  //                   prevent the UI from freezing and display a circular
+  //                   progress indicator on code size screen. Needs flutter
+  //                   web to support working with isolates. (See #33577)
+  void loadTreeFromJsonFile(DevToolsJsonFile jsonFile) async {
```

Comment:
I'm suggesting that this should return a Future, and then where you call `loadTreeFromJson` (in code_size_screen.dart) instead call `await loadTreeFromJson`

Reply:
Ah yeah, before the last commit I was calling `await loadTreeFromJson`